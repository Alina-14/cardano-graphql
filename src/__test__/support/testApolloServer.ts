import { ApolloServerBase } from 'apollo-server-core'
import * as fs from 'fs'
import * as path from 'path'
import * as depthLimit from 'graphql-depth-limit'
import { getConfig } from '../../config'

export async function testApolloServer (): Promise<ApolloServerBase> {
  process.env.HASURA_URI = 'http://localhost:8090/v1/graphql'
  const { context, resolvers } = await getConfig()
  return new ApolloServerBase({
    context,
    introspection: true,
    resolvers,
    typeDefs: fs.readFileSync(path.join(__dirname, '../../schema.graphql'), 'UTF8'),
    validationRules: [depthLimit(20)]
  })
}