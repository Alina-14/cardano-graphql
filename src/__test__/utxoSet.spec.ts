import gql from 'graphql-tag'
import { createTestClient, ApolloServerTestClient } from 'apollo-server-testing'
import { testApolloServer } from './support/testApolloServer'

describe('utxoSet', () => {
  let client: ApolloServerTestClient

  beforeEach(async () => {
    client = createTestClient(await testApolloServer())
  }, 60000)

  it('Can be scoped by address', async () => {
    const result = await client.query({
      query: gql`query {
          utxoSet(
              order_by: { address: asc }
              where: { address: { _eq:
              "DdzFFzCqrhskotfhVwhLvNFaVGpA6C4yR9DXe56oEL4Ewmze51f1uQsc1cQb8qUyqgzjUPBgFZiVbuQu7BaXrQkouyvzjYjLqfJpKG5s"
              }
              }
          ) {
              address
              value
          }
      }`
    })
    expect(result.data.utxoSet.length).toBe(2)
    expect(result).toMatchSnapshot()
  })
  it('Can be scoped by list of addresses', async () => {
    const result = await client.query({
      query: gql`query {
          utxoSet(
              order_by: { address: asc }
              where: { address: { _in: [
                  "DdzFFzCqrhskotfhVwhLvNFaVGpA6C4yR9DXe56oEL4Ewmze51f1uQsc1cQb8qUyqgzjUPBgFZiVbuQu7BaXrQkouyvzjYjLqfJpKG5s",
                  "Ae2tdPwUPEZGvXJ3ebp4LDgBhbxekAH2oKZgfahKq896fehv8oCJxmGJgLt"
              ]}}
          ) {
              address
              value
          }
      }`
    })
    expect(result.data.utxoSet.length).toBe(3)
    expect(result).toMatchSnapshot()
  })
})
