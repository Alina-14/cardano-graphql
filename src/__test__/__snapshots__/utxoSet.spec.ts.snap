// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utxoSet Can be scoped by address 1`] = `
Object {
  "data": Object {
    "utxoSet": Array [
      Object {
        "address": "DdzFFzCqrhskotfhVwhLvNFaVGpA6C4yR9DXe56oEL4Ewmze51f1uQsc1cQb8qUyqgzjUPBgFZiVbuQu7BaXrQkouyvzjYjLqfJpKG5s",
        "value": "100000000000",
      },
      Object {
        "address": "DdzFFzCqrhskotfhVwhLvNFaVGpA6C4yR9DXe56oEL4Ewmze51f1uQsc1cQb8qUyqgzjUPBgFZiVbuQu7BaXrQkouyvzjYjLqfJpKG5s",
        "value": "150000000000",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`utxoSet Can be scoped by list of addresses 1`] = `
Object {
  "data": Object {
    "utxoSet": Array [
      Object {
        "address": "Ae2tdPwUPEZGvXJ3ebp4LDgBhbxekAH2oKZgfahKq896fehv8oCJxmGJgLt",
        "value": "1153846000000",
      },
      Object {
        "address": "DdzFFzCqrhskotfhVwhLvNFaVGpA6C4yR9DXe56oEL4Ewmze51f1uQsc1cQb8qUyqgzjUPBgFZiVbuQu7BaXrQkouyvzjYjLqfJpKG5s",
        "value": "100000000000",
      },
      Object {
        "address": "DdzFFzCqrhskotfhVwhLvNFaVGpA6C4yR9DXe56oEL4Ewmze51f1uQsc1cQb8qUyqgzjUPBgFZiVbuQu7BaXrQkouyvzjYjLqfJpKG5s",
        "value": "150000000000",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
